### DISCUSSION TITLE
Tubes Contact Tutorial Using Gmsh with 3D elements

### DISCUSSION THREAD

**Discussion by drozp94** [OpenRadioss/OpenRadioss#2777] [Q&A]
Hello Community,

I am struggling with one problem and I cannot solve it for few days.
I am trying to recreate the OpenRadioss simulation example "Tubes Contact Tutorial Using Gmsh" with 3D elements instead of Shell elements.

The thing is that Gmsh export elements for volume of tubes (TET10 in my case) and also surface elements (SH3N).
If I am not defining the /PART for each SH3N group, then I get next error:

ERROR ID :    402
** ERROR IN ELEMENT PART DEFINITION
DESCRIPTION :  
   6 PART(S) REFERENCED BY ELEMENTS DO(ES) NOT EXIST. THIS MAY CAUSE OTHER ERROR MESSAGES IN THE MODEL
   /SH3N/2000001    ELEMENT(S) DEFINED IN NON EXISTING PART ID 2000001   
   /SH3N/2000002    ELEMENT(S) DEFINED IN NON EXISTING PART ID 2000002   
   /SH3N/2000003    ELEMENT(S) DEFINED IN NON EXISTING PART ID 2000003   
   /SH3N/2000004    ELEMENT(S) DEFINED IN NON EXISTING PART ID 2000004   
   /SH3N/2000005    ELEMENT(S) DEFINED IN NON EXISTING PART ID 2000005   
   /SH3N/2000006    ELEMENT(S) DEFINED IN NON EXISTING PART ID 2000006

If I am doing this, then it works, but I am confused about defining the /PART for each SH3N, because I need to add the property for this part. So what would be the thickness and if it is the correct way of doing this ?

Anybody know how to deal with 3D meshes from Gmsh ? I would really appreciate your help or even some small hints.

Thank you!
Dima

**Comment by MarianBulla** [comment]
Hello Dima,
thank you for your interest in OpenRadioss !  :-)
Regarding your question... the /PART keyword combines the definition of your property and your material with your used elements.
So, one part can contain only one sort of elements. 
You can not mix different element types within one part.
But you can reuse one property and one material within several parts. For sure, the properties only for the same type of elements.

So, there are many ways how to define contacts in OpenRadioss.

Another point is related to your tet10 elements: It is tricky for contacts, since you have to consider the mid-nodes, too. In both, the secondary nodes group AND the main-surface definition in the contact definition. For 1st-order elements (e.g. tetra4 elements or hexa8 elements) you can use something like /SURF/PART/EXT/xxx (where xxx is your surface group ID number), following with the secondary node group reusing the same contact surface with /GRNOD/SURF/xxx
For more details, please review the manual.

If you know your overall dimension of your modelled system, you can also try to fetch all surfaces within a box in space using /SURF/BOX/xxx or /SURF/BOX2/xxx.

This contact definition can be prepared within an external file, and load into the main model file with the #include keyword (e.g. "#include my_contactdefinition.inc" ).

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by drozp94** [comment]
Hello Marian,

Thanks a lot, with your message I've assembled all pieces of puzzle together. Now it's clear for me how this contact definition works in Radioss.

I've did several mistakes, I will write them for the documentation purpose, maybe it will help for somebody else dealing with the same problem.

1. So, mistake number one, and I think the most important one: In order to define the surfaces for the contact, I've used /SURF/PART/XXX, but instead of giving the Id of the 3D Part, I've provided the IDs of the 2D Shell Parts (/SH3N). I will paste an example below to better describe this mistake:
Bad definition:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/SURF/PART/1000
For Tube Contact
   2000001   2000002   2000003   2000004   2000005   2000006
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
Right definition:
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|
/SURF/PART/EXT/1000
For Tube Contact
   3000001   3000002
#---1----|----2----|----3----|----4----|----5----|----6----|----7----|----8----|----9----|---10----|

2000001 - 2000006 are SH3N elements exported from Gmsh
3000001 - 3000002 are TET elements exported from Gmsh

So I had twice defined part in the same model, once as a 3D mesh and once as a 2D mesh.


2. Mistake number two, is the export from Gmsh, I think that I should not export the surface elements at the export settings in gmsh, because those will be unused when I work with the 3D meshes. Of course in the case I would define the contact based on the element selection, then I would need to export this elements.

3. Just adding /SURF/PART/XXX from parts 3000001 and 3000002 will not solve the problem, instead /SURF/PART/EXT/XXX will solve it.


Thank you again for quick and helpful response, I will continue to learn OpenRadioss after this blocking point!
Dima
